src/k8s-engine/lima.ts

```diff
@@ -679,28 +747,129 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
+
+      if (dirInfo && dirInfo.uid === 0 && (dirInfo.mode & 2) === 0) {
+        return;
+      }
+      dirExists = true;
+    } catch (err) {
+      dirInfo = null;
+      if (err.code === 'ENOENT') {
+        dirExists = false;
+      } else {
+        console.log(`Unexpected situation with ${ RUN_LIMA_LOCATION }, stat => error ${ err }`, err);
+        throw err;
+      }
+    }
+    if (!dirInfo || !dirExists) {
+      commands.push(`mkdir -p ${ RUN_LIMA_LOCATION }`);
+      commands.push(`chmod 755 ${ RUN_LIMA_LOCATION }`);
+    } else if (dirInfo.uid !== 0) {
+      // Safe assumption that root is always user ID 0
+      commands.push(`chown -R root:wheel ${ RUN_LIMA_LOCATION }`);
+      commands.push(`chmod -R u-w ${ RUN_LIMA_LOCATION }`);
+    } else {
+      commands.push(`chmod -R u-w ${ RUN_LIMA_LOCATION }`);
+    }
+  }
+
+  /**
+   * Use the sudo-prompt library to run the script as root
+   * @param command: Name of an executable shell script (could be a binary, but no args are provided)
+   * @protected
+   */
+  protected async sudoExec(command: string) {
+    try {
+      // Try directly running the command as sudo before prompting
+      const cp = await childProcess.spawnFile('sudo', ['--non-interactive', '--', command],
+        { stdio: ['ignore', 'pipe', 'pipe'] });
+
+      if (!cp.stderr) {
+        // No need to sudo this time
+        return;
+      }
+    } catch (_) {
+    }
+    await new Promise<void>((resolve, reject) => {
+      sudo.exec(command, { name: 'Rancher Desktop', icns: resources.get('icons', 'logo-square-512.png') }, (error, stdout, stderr) => {
+        if (stdout) {
+          console.log(`Prompt for sudo: stdout: ${ stdout }`);
+        }
+        if (stderr) {
+          console.log(`Prompt for sudo: stderr: ${ stderr }`);
+        }
+        if (error) {
+          reject(error);
+        } else {
+          resolve();
+        }
       });
-    } finally {
-      await fs.promises.rm(workdir, { recursive: true });
+    });
+  }
+
+  /**
+   * Provide a default network config file with rancher-desktop specific settings.
+   *
+   * If there's an existing file, replace it if it doesn't contain a
+   * paths.varRun setting for rancher-desktop
+   */
+  protected async installCustomLimaNetworkConfig() {
+    const networkPath = path.join(paths.lima, '_config', 'networks.yaml');
+
+    try {
+      await fs.promises.access(networkPath);
+      try {
+        const data = yaml.parse(await fs.promises.readFile(networkPath, 'utf8'));
+        const runFile = data?.paths?.varRun ?? '';
+
+        if (runFile.includes('/rancher-desktop')) {
+          // Assume if there's a paths.varRun setting mentioning "rancher-desktop" there's no need to replace it.
+          return;
+        }
+      } catch (err) {
+        console.log(`Existing networks.yaml file ${ networkPath } not yaml-parsable, got error ${ err }. It will be replaced.`);
+      }
+    } catch (err) {
+      if (err.code !== 'ENOENT') {
```

Comment:
I think it would be better do something like:
```ts
try {
  const bytes = await fs.promises.readFile(networkPath, 'utf8');
} catch (err) {
  if (err.code !== 'ENOENT') …
```
Basically, catch the `ENOENT` from the actual reading, rather than an access check ahead of time; this helps with a possible race.

Having a separate catch for YAML parsing is still great, though.

Reply:
Making it simpler, the only error message we care about is if the file doesn't exist -- we report anything else, and whenever there's an error, be it problems with access or yaml-parsing, we replace the file.